### PR TITLE
Create prepare-host github actions

### PR DESCRIPTION
**Pull request by mangelajo**
Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

File: gh-actions/prepare-host/action.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        df -h
+        free -h
+
+    - name: Install WireGuard kernel module (only in WireGuard jobs)
+      shell: bash
+      run: |
+        if [[ "${{ inputs.cabledriver }}" == "wireguard" ]]; then
+          sudo apt install -y linux-headers-$(uname -r) wireguard
+          sudo modprobe wireguard
+        fi
```
Comment (mkolesnik) [inline gh-actions/prepare-host/action.yaml]:
```suggestion
        [[ "${{ inputs.cabledriver }}" == "wireguard" ]] || exit
        sudo apt install -y linux-headers-$(uname -r) wireguard
        sudo modprobe wireguard
```

File: gh-actions/prepare-host/action.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    description: 'Install any system modules for specific cable drivers'
+    required: false
+    default: ''
+runs:
+  using: "composite"
+  steps:
+    - name: Reclaim free space
+      shell: bash
+      run: |
+        # cleanup 30GB+ of tools we don't need for our CI
```
Comment (mkolesnik) [inline gh-actions/prepare-host/action.yaml:14]:
```suggestion
        # Clean up tools we don't need for our CI to free up more space on the hosted runner
```

**Review by mkolesnik — CHANGES_REQUESTED** [review]
I still prefer that we extract the entire e2e with the prepare, not just the prepare, as its an implementation detail of running e2e tests and shouldnt be exposed externally

File: gh-actions/prepare-host/action.yaml (hunk 1 above)
Comment (mangelajo) [reply gh-actions/prepare-host/action.yaml]: I'm trying nested groups to see if it works... to make more informative output. If it doesn't work I'll use this suggestion.

File: gh-actions/prepare-host/action.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        echo "::group::Install any cable driver specific modules"
+
+        if [[ "${{ inputs.cabledriver }}" == "wireguard" ]]; then
+          echo "::group::Installing wireguard modules"
+          sudo apt install -y linux-headers-$(uname -r) wireguard
+          sudo modprobe wireguard
+          echo "::endgroup::"
+        fi
+
+        echo "::endgroup::"
```
Comment (mkolesnik) [inline gh-actions/prepare-host/action.yaml:31]:
```suggestion
        [[ "${{ inputs.cabledriver }}" == "wireguard" ]] || exit

        echo "::group::Installing wireguard modules"
        sudo apt install -y linux-headers-$(uname -r) wireguard
        sudo modprobe wireguard
        echo "::endgroup::" 
```

File: gh-actions/prepare-host/action.yaml (hunk 1 above)
Comment (mangelajo) [reply gh-actions/prepare-host/action.yaml]: https://github.com/submariner-io/shipyard/pull/422/checks?check_run_id=1769053211#step:3:17 it doesn't do nesting properly, so... I'll apply

**Review by mkolesnik — CHANGES_REQUESTED** [review]

File: gh-actions/prepare-host/action.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    - shell: bash
+      run: |
+        echo "::group::Report available RAM"
+        free -h
+        echo "::endgroup::"
+
+    - shell: bash
+      run: |
+        echo "::group::Install any cable driver specific modules"
```
Comment (mkolesnik) [inline gh-actions/prepare-host/action.yaml:27]:
```suggestion
```

We can just not print it, since nothing happens and the group is empty

Comment (mangelajo) [reply gh-actions/prepare-host/action.yaml:27]: +1

Comment (mangelajo) [reply gh-actions/prepare-host/action.yaml]: Done.

File: gh-actions/prepare-host/action.yaml (hunk 3 above)
Comment (mangelajo) [reply gh-actions/prepare-host/action.yaml]: Done

File: gh-actions/prepare-host/action.yaml (hunk 2 above)
Comment (mangelajo) [reply gh-actions/prepare-host/action.yaml]: Done.

File: gh-actions/prepare-host/action.yaml (hunk 1 above)
Comment (mangelajo) [reply gh-actions/prepare-host/action.yaml]: Done.

**Review by mkolesnik — APPROVED** [review]
Looks good as a tart, lets merge this and I'll add the E2E parts

**Review by skitt — APPROVED** [review]

**Pull request merged**